python/cudf/cudf/tests/test_cuda_array_interface.py

```diff
@@ -193,10 +195,11 @@ def test_cuda_array_interface_pytorch():
 
     assert_eq(got, cudf.Series(buffer, dtype=np.bool_))
 
-    index = cudf.Index([], dtype="float64")
-    tensor = torch.tensor(index)
-    got = cudf.Index(tensor)
-    assert_eq(got, index)
+    # TODO: This test fails with PyTorch 2. Is it still expected to be valid?
+    # index = cudf.Index([], dtype="float64")
```

Comment:
This test fails with an empty Index, because the data pointer is `0`. I don't think that an empty Index object is backed by any device data, but it appears that it should be valid to use it in `__cuda_array_interface__`. [The docs state](https://numba.readthedocs.io/en/stable/cuda/cuda_array_interface.html) "For zero-size arrays, use 0 here."

The data returned by `index.__cuda_array_interface__` is:
```python
{'shape': (0,), 'strides': (8,), 'typestr': '<f8', 'data': (0, False), 'version': 1}
```

The error states:
```
RuntimeError: The specified pointer resides on host memory and is not registered with any CUDA device.
```

It seems the error is coming from here:
https://github.com/pytorch/pytorch/blob/2d17230212968e1d1968701ed4d2ea1bc2bf4c22/aten/src/ATen/cuda/CUDADevice.h#L17

I wonder if that code should check to see if the size is zero before verifying that the pointer is device-accessible.

Reply:
Yeah I think this is a bug in torch. @shwina @leofang @gmarkall would you agree? If so, we should report it upstream.

Reply:
@vyasr See https://github.com/pytorch/pytorch/issues/98133, I linked it in this PR before merging.

Reply:
Thanks!

Reply:
Right, @vyasr you should bug the DLFW team internally to get attention. Seems like we reported the bug and then moved on lol

Reply:
Wait, no, the bug fix was sent immediately after the report, but for some reason it didn't get attention and got closed instead: https://github.com/pytorch/pytorch/pull/98134